tests/integration_tests/test_ds_identify.py

```diff
@@ -0,0 +1,38 @@
+"""test that ds-identify works as expected"""
+
+import json
+
+from tests.integration_tests.instances import IntegrationInstance
+from tests.integration_tests.integration_settings import OS_IMAGE, PLATFORM
+from tests.integration_tests.util import verify_clean_log, wait_for_cloud_init
+
+DATASOURCE_LIST_FILE = "/etc/cloud/cloud.cfg.d/90_dpkg.cfg"
+MAP_PLATFORM_TO_DATASOURCE = {
+    "lxd_container": "lxd",
+    "lxd_vm": "lxd",
+    "qemu": "nocloud",
+    "ec2": "aws",
+    "oci": "oracle",
+}
+
+
+def test_ds_identify(client: IntegrationInstance):
+    """Verify that ds-identify works correctly
+
+    Deb packaging often a datasource_list with a single datasource, which
+    bypasses ds-identify. This tests works by removing this file and verifying
+    that cloud-init doesn't experience issues.
+    """
+    assert client.execute(f"rm {DATASOURCE_LIST_FILE}")
+    assert client.execute("cloud-init clean --logs")
+    client.restart()
+    wait_for_cloud_init(client)
+    verify_clean_log(client.execute("cat /var/log/cloud-init.log"))
+    assert client.execute("cloud-init status --wait")
+
+    datasource = MAP_PLATFORM_TO_DATASOURCE.get(PLATFORM, PLATFORM)
+    if "lxd" == datasource and "focal" == OS_IMAGE:
+        datasource = "nocloud"
+    cloud_id = client.execute("cloud-id")
```

Comment:
is this any functionally different than:
```
json.loads(
        client.execute("cloud-init status --format json").stdout
    ).get("datasource")
```

or is it the same under the hood and just a much cleaner way of doing the same thing?

Reply:
Now that i see that you added EC2 back in as "aws" I assume the cloud-id _is_ functionally different? It prevents the issue of the various EC2 datasource names?

Reply:
Yeah, the do pretty different things - "datasource" reports the name of the class that was used without the prefix (on ec2, DataSourceEc2Local gets shortened to "ec2local"). `cloud-id` is more user-facing but can come directly from the cloud's IMDS (and it falls back [to `dsname`](https://docs.cloud-init.io/en/latest/reference/datasource_dsname_map.html)).